core/backend/src/ECSqlStatement.ts

```diff
@@ -689,6 +689,17 @@ export interface ECSqlColumnInfo {
    */
   getPropertyName(): string;
 
+  /** Gets the name of the original property for the column, if backed by a property.
+   * > Other than getPropertyName(), this ignores aliases and allows getting the name
+   * > of the property which is being used for the column. A column may not be backed
+   * > by a property, so check hasOriginProperty() before using this.
+   * @throws IModelError in case of errors (e.g. column not backed by a property)
```

Comment:
Why does it not simply returned undefined if not backed by a property?

Reply:
From a typescript perspective undefined seems better. This just felt more in line with the rest of our node addon API.
We do have some spots where we return undefined though. For example GetEnum().
You prefer undefined?

Reply:
Alright, I decided to change the signature, withdrawing this PR for now, I'll republish it for the imodel02 branch later.

Reply:
Could just have typescript catch this specific error and return undefined instead of changing native.